src/extensions/windows/Microsoft.DotNet.UpgradeAssistant.Extensions.Windows/UWPtoWinAppSDKUpgrade/WinUIAppWIndowAnalyzer.cs

```diff
@@ -0,0 +1,184 @@
+                    ("Microsoft.UI.Windowing", "AppWindow", "CloseAsync"),
+                    ("*", "*", "Destroy"))
+            }
+        };
+
+        private void AnalyzeIdentifierName(SyntaxNodeAnalysisContext context)
+        {
+            var node = (IdentifierNameSyntax)context.Node;
+            foreach (var api in TypeConversions)
+            {
+                var (fromNamespace, fromType) = api.Value.Key;
+                if (node.Identifier.ValueText.ToString() == fromType)
+                {
+                    if (node.Parent.IsKind(SyntaxKind.QualifiedName))
+                    {
+                        if (((QualifiedNameSyntax)node.Parent).Left.ToString() == fromNamespace)
+                        {
+                            ReportDiagnostic(api.Key, node.Parent.GetLocation());
+                            return;
+                        }
+
+                        continue;
+                    }
+
+                    if (node.Parent.IsKind(SyntaxKind.SimpleMemberAccessExpression))
+                    {
+                        if (node.Parent.ToString() == $"{fromNamespace}.{fromType}")
+                        {
+                            ReportDiagnostic(api.Key, node.Parent.GetLocation());
+                            return;
+                        }
+
+                        if (!node.Parent.ToString().StartsWith(fromType, StringComparison.Ordinal))
+                        {
+                            return;
+                        }
+                    }
+
+                    if (UWPToWinUIHelpers.GetAllImportedNamespaces(context).Contains(fromNamespace))
+                    {
+                        ReportDiagnostic(api.Key, node.GetLocation());
+                        return;
+                    }
+                }
+            }
+
+            void ReportDiagnostic(string apiId, Location location)
+            {
+                var diagnostic = Diagnostic.Create(RuleAppWindowNamespace, location, ImmutableDictionary.Create<string, string?>()
+                        .Add("apiId", apiId), node.GetText().ToString());
+                context.ReportDiagnostic(diagnostic);
+            }
+        }
+
+        private void AnalyzeInvocationExpression(SyntaxNodeAnalysisContext context)
+        {
+            var node = (InvocationExpressionSyntax)context.Node;
+
+            var memberAccessExpression = node.Expression;
+            if (memberAccessExpression is null || !memberAccessExpression.IsKind(SyntaxKind.SimpleMemberAccessExpression))
+            {
+                return;
+            }
+
+            var varName = ((MemberAccessExpressionSyntax)memberAccessExpression).Expression;
+            var type = context.SemanticModel.GetTypeInfo(varName);
+            var allNamespaces = UWPToWinUIHelpers.GetAllImportedNamespaces(context);
+            var memberAccessString = memberAccessExpression.ToString();
```

Comment:
best not to call ToString - this ends up calling it for all simplememberaccessexpressions.

Reply:
Any alternate ways of doing this then?